sotodlib/wiregrid/calibrate.py

```diff
@@ -0,0 +1,404 @@
+        pass
+    ts_step_start = aman.timestamps[step_start]
+    ts_step_stop = aman.timestamps[step_stop]
+    ang_av = np.array(ang_av)
+    ang_std = np.array(ang_std)
+    #
+    return (ts_step_start, ts_step_stop), (ang_av, ang_std)
+
+# Wrap the QU response during the operation of the Grid Loader
+def wg_wrap_QU(aman, stopped_time, thresholds=None):
+    """
+    Wrap QU signal by the wire grid.
+
+    Parameters:
+    - aman : AxisManager
+    - stopped_time : the stopped time of your target operation
+    - threshold : the thresholds on the encoder counts.
+        the first element is the upper bound for the static state,
+        and the second element is the lower bound for the difference of the first
+
+    Return:
+    - aman : AxisManger
+        This includes the characterics of the wire grid operation and the Q/U signal
+        related with it.
+        flag_step_start : the start time stamps for the steps
+        flag_step_stop : the stop time stamps for the steps
+        enc_ang_deg : encouder counts in degreee for each step
+        enc_ang_std : the standard deviations of the encoder counts for steps
+        Q, U : Q (and U) signal for steps
+        Qerr, Uerr : the standard deviations of Q (and U) signal for steps
+    """
+    ts_step, wg_deg = _detect_steps(aman, stopped_time, thresholds)
+    ts_instep = []
+    step_Q = []
+    step_U = []
+    step_Qerr = []
+    step_Uerr = []
+    for _i in range(len(wg_deg[0])):
+        instep = np.where(ts_step[0][_i] < aman.timestamps, True, False)
+        instep = np.where(aman.timestamps < ts_step[1][_i+1], instep, False)
+        #
+        ts_instep.append(np.average(aman.timestamps[instep]))
+        step_Q.append(np.average(aman.demodQ, axis=1, weights=instep))
+        step_U.append(np.average(aman.demodU, axis=1, weights=instep))
+        step_Qerr.append(np.std(aman.demodQ[:, instep == True], axis=1))
+        step_Uerr.append(np.std(aman.demodU[:, instep == True], axis=1))
+        pass
+    ts_instep = np.array(ts_instep)
+    step_Q = np.array(step_Q)
+    step_U = np.array(step_U)
+    step_Qerr = np.array(step_Qerr)
+    step_Uerr = np.array(step_Uerr)
+    #
+    aman.wg.wrap('flag_step_start', ts_step[0])
+    aman.wg.wrap('flag_step_stop', ts_step[1])
+    aman.wg.wrap('enc_ang_deg', wg_deg[0])
+    aman.wg.wrap('enc_ang_std', wg_deg[1])
+    aman.wg.wrap('ts_instep', ts_instep)
+    aman.wg.wrap('Q', step_Q, [(1, 'dets')])
+    aman.wg.wrap('U', step_U, [(1, 'dets')])
+    aman.wg.wrap('Qerr', step_Qerr, [(1, 'dets')])
+    aman.wg.wrap('Uerr', step_Uerr, [(1, 'dets')])
+    return aman
+
+def _get_initial_param_circle(x):
+    """
+    Get the initial parameters as inputs of the circle fitting in Q-U plane
+
+    Paramter:
+    - x : assuming ndarray of Q and U signal with 16 steps in nominal, np.array([Q(16), U(16)])
+
+    Return:
+    - params : the initial paramters for the circle fit
+    """
+    if len(np.shape(x)) == 2:
```

Comment:
Explain in docstring what the len(x) == 2 and == 3 cases correspond to.

Reply:
Well, this is a little unclear section. I will give some explanations. Thank you.